force_nevergrad/mco/ng_mco_communicator.py

```diff
@@ -0,0 +1,90 @@
+#  (C) Copyright 2010-2020 Enthought, Inc., Austin, TX
+#  All rights reserved.
+
+import sys
+import re
+
+from force_bdss.api import (
+    BaseMCOCommunicator,
+    DataValue,
+    FixedMCOParameter,
+    RangedMCOParameter,
+    RangedVectorMCOParameter,
+    ListedMCOParameter,
+    CategoricalMCOParameter
+)
+
+
+class NevergradMCOCommunicator(BaseMCOCommunicator):
+    """ Command-line evaluation.
+
+    Examples
+    --------
+    Evaluate the point (-1.0, 1.0) and write to output.txt,
+
+    echo 1.0,-1.0 | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    Evaluate the point on the first line of input.txt and write to output.txt,
+
+    Windows-native:
+    type input.txt | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    Mac/bash:
+    cat input.txt | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    If you are in the EDM force-py36 environment, then just leave out the
+    'edm run -e force-py36 --' part of the command.
+
+    Notes
+    -----
+    Can only evaluate a single point from stdin. If you want to evaluate
+    multiple points listed in a file, then you must write a script
+    that loops through the file's lines (one line per point) and evaluates
+    each one in turn. cf. evalf.cmd for a windows-native script.
+    """
+
+    def receive_from_mco(self, model):
+        """ Get the parameter values to evaluate from stdin.
+        """
+        # Read in a line of points to evaluate.
+        # Can be tab or comma delimited.
+        line = sys.stdin.readline()
+        data = re.split(r'[,\s]+', line)
+
+        # Get the parameter values based on parameterization and stdin
+        inputs = []
+        for i, param in enumerate(model.parameters):
+
+            # default value of parameter
+            if type(param) == FixedMCOParameter:
+                v = param.value
+            elif type(param) == RangedMCOParameter:
+                v = param.initial_value
+            elif type(param) == RangedVectorMCOParameter:
+                v = param.initial_value
+            elif type(param) == ListedMCOParameter:
+                v = param.levels[0]
+            elif type(param) == CategoricalMCOParameter:
+                v = param.categories[0]
+            else:
+                v = 0.0
```

Comment:
We could use `isinstance` here and class inheritance to tidy things up a bit
```suggestion
            # default value of parameter
            if isinstance(param, FixedMCOParameter):
                v = param.value
            elif isinstance(param, RangedMCOParameter):
                v = param.initial_value
            elif isinstance(param, ListedMCOParameter):
                v = param.levels[0]
            elif isinstance(param, CategoricalMCOParameter):
                v = param.categories[0]
            else:
                v = 0.0
```

Reply:
This is the same issue as the "translators" for the nevergrad engine/optimizer - isinstance matches subclasses and so makes  problems as some of the parameters are derived from each other. In particular it causes unit tests to skip some elifs, so the coverage doesn't pass.

Reply:
Sure, but presumably in this case we don't need to distinguish between `RangedMCOParameter` and `RangedVectorMCOParameter`?.

Reply:
> Looks good, I only have some only some minor comments regarding the implementation.
> 
> If we include the `shell_examples` folder in the repository it should go inside the `force_nevergrad/tests/fixtures/` directory, to show that it is not part of the main package. It would also be nice to include bash scripts (`opts.sh`, `evalf.sh`) that perform the equivalent operations.

I think I need to make another installation of edm, with linux/ubuntu (at least to test the scripts). Is there another way?

Reply:
I don't think so, since you'll need to generate the appropriate binaries to run on Linux. 

EDM should work in windows sub system, but you might come across some difficulties using the `edm shell` command. I think `edm run` is ok though.